packages/host/src/components/Footer/StoreSelector.tsx

```diff
@@ -8,51 +8,64 @@ import {
   usePaperClickPopover,
   useTranslation,
   useNavigate,
+  useUserStores,
 } from '@openmsupply-client/common';
 import { AppRoute } from '@openmsupply-client/config';
-import { StoreRowFragment, useStores } from '@openmsupply-client/system';
+import { UserStoreNodeFragment } from 'packages/common/src/authentication/api/operations.generated';
+
+const POPPER_HEIGHT = 400;
 
 export const StoreSelector: FC = ({ children }) => {
-  const { store, setStore } = useAuthContext();
+  const { store, setStore, token } = useAuthContext();
   const navigate = useNavigate();
   const { hide, PaperClickPopover } = usePaperClickPopover();
-  const { data, isLoading } = useStores();
+  const { data, isLoading } = useUserStores(token);
   const t = useTranslation('app');
 
-  const storeSorter = (a: StoreRowFragment, b: StoreRowFragment) => {
-    if (a.code < b.code) return -1;
-    if (a.code > b.code) return 1;
+  const storeSorter = (a: UserStoreNodeFragment, b: UserStoreNodeFragment) => {
+    if (a.name < b.name) return -1;
+    if (a.name > b.name) return 1;
     return 0;
   };
-  const stores = (data?.nodes ?? []).sort(storeSorter);
+  const stores = data?.sort(storeSorter) || [];
 
-  if (!store?.code) return null;
+  if (!store?.name) return null;
 
   if (stores.length < 2) return <>{children}</>;
 
   const storeButtons = stores.map(s => (
     <FlatButton
-      label={s.code}
+      label={s.name}
       disabled={s.id === store.id}
       onClick={() => {
         setStore(s);
         hide();
         navigate(AppRoute.Dashboard);
       }}
       key={s.id}
+      sx={{
+        justifyContent: 'flex-start',
+        whiteSpace: 'nowrap',
+      }}
     />
   ));
   return (
     <PaperClickPopover
       placement="top"
-      width={250}
+      width={300}
+      height={POPPER_HEIGHT}
       Content={
         <PaperPopoverSection label={t('select-store')}>
           {isLoading ? (
             <CircularProgress size={12} />
           ) : (
             <Box
-              style={{ maxHeight: '200px', overflowY: 'auto' }}
+              style={{
+                overflowX: 'hidden',
+                textOverflow: 'ellipsis',
```

Comment:
oh. yeah - the overflow isn't working and I couldn't see how to make it work as I expected it to. gave up in the end 🤷

Reply:
It works if you move the horizontal overflow style properties to the `<FlatButton>` component instead. So the `sx` property on `FlatButton` should be:
```
sx={{
        whiteSpace: 'nowrap',
        overflowX: 'hidden',
        textOverflow: 'ellipsis',
        display: 'block',
        textAlign: 'left',
      }}
```
`display: block` is just to override the default `display: inline-flex` which messes up the ellipsis.

Reply:
thanks. yeah, flex 🙄